### PR TITLE
Added ES module bundle to the auth package

### LINKED ISSUE #818: Auth: ES modules build
### Environment

  * Operating System version: any
  * Firebase SDK version: 5.0.2
  * Firebase Product: auth

### The problem

`auth` package does not include an ES modules build

#### Steps to reproduce:

Install `@firebase/app` and use it in a project aimed for module-aware browsers, e.g. Polymer 3 application. Observe an error in the console:
```
Uncaught ReferenceError: require is not defined
    at auth.js:2
    [...]
```

The issue is caused by a lack of ES module build of the library (no valid output file and no `module` field in a `package.json`.

#### Relevant Code:
```javascript
import firebase from '@firebase/app';
import '@firebase/auth';
```

### PR DESCRIPTION
**Pull request by merlinnot**
Resolves #818 

There is no changelog entry 'cause I couldn't find it ;)

### CODE REVIEW

**Comment by merlinnot** [comment]
Travis probably needs a retry.

**Comment by merlinnot** [comment]
May I kindly ask for a review?

**Comment by bojeil-google** [comment]
Hey @merlinnot, thanks for your contribution. I reran the tests several times and they are all failing. Please fix and I can help review.

**Comment by merlinnot** [comment]
I will take a look in the evening.

**Comment by merlinnot** [comment]
Travis build: https://github.com/firebase/firebase-js-sdk/issues/818#issuecomment-390629909

**Comment by merlinnot** [comment]
@bojeil-google test are passing, I'd appreciate a review :)

File: packages/auth/buildtools/run_tests.sh [hunk 1]
```diff
@@ -47,13 +47,28 @@
 #
 # Travis will run `npm test -- --saucelabs`.
 
+# Since yarn workspaces might hoist our packages, we have to fallback to
+# ../../node_modules when we want to execute a script.
+declare -a nodeModulesBasedirs=(
+  "./node_modules/.bin"
+  "../../node_modules/.bin"
+)
+function evalModule {
```
Comment (bojeil-google) [inline packages/auth/buildtools/run_tests.sh]: Declare javadoc or some guidance on this function's usage/signature.

File: packages/auth/buildtools/run_tests.sh [hunk 2]
```diff
@@ -47,13 +47,28 @@
 #
 # Travis will run `npm test -- --saucelabs`.
 
+# Since yarn workspaces might hoist our packages, we have to fallback to
+# ../../node_modules when we want to execute a script.
+declare -a nodeModulesBasedirs=(
+  "./node_modules/.bin"
+  "../../node_modules/.bin"
+)
```
Comment (bojeil-google) [inline packages/auth/buildtools/run_tests.sh]: Add new line after this

File: packages/auth/gulpfile.js [hunk 3]
```diff
@@ -19,52 +19,82 @@ const closureCompiler = require('gulp-closure-compiler');
...
 const OPTIMIZATION_LEVEL = 'ADVANCED_OPTIMIZATIONS';
 
 // For minified builds, wrap the output so we avoid leaking global variables.
-const OUTPUT_WRAPPER = `(function() {
-  var firebase = require('@firebase/app').default;
-  %output%
-}).call(typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : {});`;
+const CJS_WRAPPER_PREFIX = `(function() {var firebase = require('@firebase/app').default;`;
+const EMS_WRAPPER_PREFIX = `import firebase from '@firebase/app';(function() {`;
+const WRAPPER_SUFFIX = `}).call(typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : {});`;
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: Line is too long. Split into multiple lines.

File: packages/auth/gulpfile.js [hunk 4]
```diff
@@ -75,11 +105,13 @@ gulp.task('clean', done => del(['dist/*', 'dist'], done));
 gulp.task('serve', () => {
   const app = express();
 
-  app.use('/node_modules', express.static(path.resolve(__dirname, '../../node_modules')));
+  app.use(
+    '/node_modules',
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: indent 2 additional spaces. Style guide recommendations is to indent 4 spaces for function arguments that spill over to next line.

File: packages/auth/gulpfile.js [hunk 5]
```diff
@@ -75,11 +105,13 @@ gulp.task('clean', done => del(['dist/*', 'dist'], done));
...
+    express.static(path.resolve(__dirname, '../../node_modules'))
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: indent 2 additional spaces.

File: packages/auth/gulpfile.js [hunk 6]
```diff
@@ -19,52 +19,82 @@ const closureCompiler = require('gulp-closure-compiler');
...
 // The path to Closure Compiler.
-const COMPILER_PATH = `${path.dirname(require.resolve('google-closure-compiler/package.json'))}/compiler.jar`;
+const COMPILER_PATH = `${path.dirname(
+  require.resolve('google-closure-compiler/package.json')
+)}/compiler.jar`;
 
-const closureLibRoot = path.dirname(require.resolve('google-closure-library/package.json'));
-// Builds the core Firebase-auth JS.
+const closureLibRoot = path.dirname(
+  require.resolve('google-closure-library/package.json')
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: indent 2 additional spaces.

File: packages/auth/gulpfile.js [hunk 7]
```diff
@@ -19,52 +19,82 @@ const closureCompiler = require('gulp-closure-compiler');
...
-          path.resolve(__dirname, '../firebase/externs/firebase-error-externs.js'),
-          path.resolve(__dirname, '../firebase/externs/firebase-app-internal-externs.js')
-        ],
-        language_out: 'ES5',
-        only_closure_dependencies: true,
-        output_wrapper: OUTPUT_WRAPPER
-      }
-    }))
+    .pipe(
+      closureCompiler({
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: indent entire block 2 additional spaces.

File: packages/auth/gulpfile.js [hunk 8]
```diff
@@ -19,52 +19,82 @@ const closureCompiler = require('gulp-closure-compiler');
...
+        fileName: 'unwrapped.js',
+        compilerFlags: {
+          closure_entry_point: 'fireauth.exports',
+          compilation_level: OPTIMIZATION_LEVEL,
+          externs: [
+            'externs/externs.js',
+            'externs/grecaptcha.js',
+            'externs/gapi.iframes.js',
+            path.resolve(
+              __dirname,
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: Both arguments need 2 additional spaces, here and below for each `path.resolve(`.

File: packages/auth/gulpfile.js [hunk 9]
```diff
@@ -19,52 +19,82 @@ const closureCompiler = require('gulp-closure-compiler');
...
 // For minified builds, wrap the output so we avoid leaking global variables.
-const OUTPUT_WRAPPER = `(function() {
-  var firebase = require('@firebase/app').default;
-  %output%
-}).call(typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : {});`;
+const CJS_WRAPPER_PREFIX = `(function() {var firebase = require('@firebase/app').default;`;
+const EMS_WRAPPER_PREFIX = `import firebase from '@firebase/app';(function() {`;
+const WRAPPER_SUFFIX = `}).call(typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : {});`;
+
+const wrap = through(function(file) {
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: Add comment or javadoc to explain what this function does.

**Comment by merlinnot** [comment]
Thanks for the review, gonna take care of it tomorrow.

**Comment by merlinnot** [comment]
@bojeil-google sorry for not aligning with a style guide, it's hard to match all of the criteria when you use other standards on a daily basis. I've fixed the issues you've mentioned and one additional call to align with the rule you've pointed out.

File: packages/auth/gulpfile.js [hunk 10]
```diff
@@ -28,10 +28,18 @@ const File = require('vinyl');
 const OPTIMIZATION_LEVEL = 'ADVANCED_OPTIMIZATIONS';
 
 // For minified builds, wrap the output so we avoid leaking global variables.
-const CJS_WRAPPER_PREFIX = `(function() {var firebase = require('@firebase/app').default;`;
+const CJS_WRAPPER_PREFIX =
+  `(function() {var firebase = require('@firebase/app').default;`;
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: Add 2 more spaces here. (if right hand side of = is on second line)
```
const CJS_WRAPPER_PREFIX =
    'value';
```

File: packages/auth/gulpfile.js [hunk 11]
```diff
@@ -28,10 +28,18 @@ const File = require('vinyl');
...
 const EMS_WRAPPER_PREFIX = `import firebase from '@firebase/app';(function() {`;
-const WRAPPER_SUFFIX = `}).call(typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : {});`;
+const WRAPPER_SUFFIX =
+  `}).call(typeof global !== 'undefined' ? ` +
+  `global : typeof self !== 'undefined' ? ` +
+  `self : typeof window !== 'undefined' ? window : {});`;
```
Comment (bojeil-google) [inline packages/auth/gulpfile.js]: Same here for above block:
Add 2 more spaces here. (if right hand side of = is on second line)
```
const WRAPPER_SUFFIX =
    'value1' +
    'value2';
```

**Comment by merlinnot** [comment]
It is actually documented [here](https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/STYLEGUIDE.md), the thing that is missing is an auto-formatting tool that would check for all of these rules. It's enough to have a documentation for people who work on the project on a daily basis, but it's hard to expect from external contributors to learn all of theses rules to make a few small PRs, it's just a waste of time.

A tool like Prettier (aren't you already using it in other packages?) or clang would do.

Thank you for the approval!

**Pull request merged**